configuration/postgres/postgres.go

```diff
@@ -0,0 +1,417 @@
+	}
+	pool, err := pgxpool.NewWithConfig(ctx, config)
+	if err != nil {
+		return nil, fmt.Errorf("postgres configuration store connection error : %s", err)
+	}
+	pingErr := pool.Ping(ctx)
+	if pingErr != nil {
+		return nil, fmt.Errorf("postgres configuration store ping error : %s", pingErr)
+	}
+	return pool, nil
+}
+
+func buildQuery(req *configuration.GetRequest, configTable string) (string, []interface{}, error) {
+	var query string
+	var params []interface{}
+	if len(req.Keys) == 0 {
+		query = "SELECT * FROM " + configTable
+	} else {
+		var queryBuilder strings.Builder
+		queryBuilder.WriteString("SELECT * FROM " + configTable + " WHERE KEY IN (")
+		var paramWildcard []string
+		paramPosition := 1
+		for _, v := range req.Keys {
+			paramWildcard = append(paramWildcard, "$"+strconv.Itoa(paramPosition))
+			params = append(params, v)
+			paramPosition++
+		}
+		queryBuilder.WriteString(strings.Join(paramWildcard, " , "))
+		queryBuilder.WriteString(")")
+		query = queryBuilder.String()
+
+		if len(req.Metadata) > 0 {
+			var s strings.Builder
+			i, j := len(req.Metadata), 0
+			s.WriteString(" AND ")
+			for k, v := range req.Metadata {
+				temp := "$" + strconv.Itoa(paramPosition) + " = " + "$" + strconv.Itoa(paramPosition+1)
+				s.WriteString(temp)
+				params = append(params, k, v)
+				paramPosition += 2
+				if j++; j < i {
+					s.WriteString(" AND ")
+				}
+			}
+			query += s.String()
+		}
+	}
+	return query, params, nil
+}
+
+func (p *ConfigurationStore) isSubscriptionActive(req *configuration.SubscribeRequest) (string, bool) {
+	for _, channel := range req.Metadata {
+		for key2, sub := range p.ActiveSubscriptions {
+			if res := strings.EqualFold(channel, key2); res {
+				return sub.uuid, true
+			}
+		}
+	}
+	return " ", false
+}
+
+func (p *ConfigurationStore) isSubscribed(subscriptionID string, channel string, key string) bool {
+	if val, yes := p.ActiveSubscriptions[channel]; yes && val.uuid == subscriptionID && slices.Contains(val.keys, key) {
+		return true
+	}
+	return false
+}
+
+func validateInput(keys []string) error {
+	for _, key := range keys {
+		if !allowedChars.MatchString(key) {
+			return fmt.Errorf("invalid key : '%v'", key)
+		}
+	}
+	return nil
+}
+
+func (p *ConfigurationStore) subscribeToChannel(ctx context.Context, pgNotifyChanList []string, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
+	p.configLock.Lock()
+	defer p.configLock.Unlock()
+	var subscribeID string
+	for _, channel := range pgNotifyChanList {
+		pgNotifyCmd := fmt.Sprintf(listenTemplate, channel)
+		if sub, isActive := p.isSubscriptionActive(req); isActive {
+			if oldStopChan, ok := p.subscribeStopChanMap[sub]; ok {
+				close(oldStopChan.(chan struct{}))
+				delete(p.subscribeStopChanMap, sub)
+				delete(p.ActiveSubscriptions, channel)
+			}
+		}
+		stop := make(chan struct{})
+		subscribeID = uuid.New().String()
```

Comment:
Are you referring to NewUUID? Recommedation is to use New method. - https://pkg.go.dev/github.com/google/UUID#NewUUID

Reply:
No I'm referring to NewRandom: https://pkg.go.dev/github.com/google/UUID#NewRandom 

Internally, uuid.New uses NewRandom but discards the error, and panics in case of error. That's not ideal. https://github.com/google/UUID/blob/v1.0.0/version4.go#L13

Reply:
Updated